### PR TITLE
Feat/accept entry

### PR DESCRIPTION
**Pull request by LeonNimoy**
Updates:
-Create accept:entry action;
-Install Prettier.

### CODE REVIEW

**Comment by LeonNimoy** [comment]
Please, @rjrodger,  if you see suitable, try to install this package on emacs: **M-x package-install prettier** and add this hook **(add-hook 'after-init-hook #'global-prettier-mode)**. I think it might solve this problem.

Also, I just updated this PR. Now, I added a format command to npm. Therefore, we can format our files through CLI: 
```bash
npm run format
```
As I Included this command in our build pipeline. Even if the file isn't formatted in the local machine, it should always format when build process is finished.

**Comment by rjrodger** [comment]
please restore tsfmt.json - this is used by emacs ts mode

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Ignore artifacts:
```
Comment (rjrodger) [inline .prettierignore:1]: why is this needed? we are only running prettier on ts files?

**Review by rjrodger — CHANGES_REQUESTED** [review]

File: .prettierrc.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "singleQuote": true,
+  "semi": false,
+  "trailingComma": "none"
```
Comment (rjrodger) [inline .prettierrc.json:4]: trailingCommas are a good thing - allows for easy cut and paste re-ordering - this is why they have been added to JS

File: src/refer-doc.ts [hunk 3]
```diff
@@ -1,16 +1,13 @@
 /* Copyright © 2022 Seneca Project Contributors, MIT License. */
 
-
 const docs = {
-
   actCreateEntry: {
-    desc: 'Create referral entry.',
+    desc: "Create referral entry.",
```
[the remark this answers is not in the document]
Comment (LeonNimoy) [reply src/refer-doc.ts]: Understood, thank you for the reference. I managed to make some recent updates. And move back to use "single quotes", instead of double. When you can, take a look again, please.

File: src/refer-doc.ts [hunk 4]
```diff
@@ -1,16 +1,13 @@
...
   },
+};
```
[the remark this answers is not in the document]
Comment (LeonNimoy) [reply src/refer-doc.ts]: Indeed. I'm gonna follow the seneca convention. I managed to make some recent updates, and removed the semicolons. When you get some time, take a look again, please.

File: src/refer.ts [hunk 5]
```diff
@@ -1,112 +1,122 @@
...
+      occur: [occur],
+    };
   }
 
+  async function actAcceptEntry(this: any, msg: any) {
+    const seneca = this;
 
+    const entryList = await seneca
+      .entity("refer/entry")
+      .list$({ user_id: msg.user_id });
```
[the remark this answers is not in the document]
Comment (LeonNimoy) [reply src/refer.ts]: Precisely. Changes on commit: 35fb7af75524c854bd6c1829f8b180f03f674f45

File: src/refer.ts [hunk 6]
```diff
@@ -1,112 +1,122 @@
...
+    const entry = entryList[0];
 
+    const occur = await seneca.entity("refer/occur").save$({
```
[the remark this answers is not in the document]
Comment (LeonNimoy) [reply src/refer.ts]: Indeed, changes on commits: https://github.com/senecajs/seneca-refer/commit/4f4b2a0cc5e697f7b5058bd7df2d6c638fbf3604 and https://github.com/senecajs/seneca-refer/commit/2496e835b24d5114003bcb7ac77845544cf5e0f8

Its a simple validation, but my intention is to create an abstraction, as the actions begin to grow. Although, I'm totally willing to do any more changes, now.

File: .prettierignore (hunk 1 above)
Comment (rjrodger) [reply .prettierignore:1]: ok we can try it

File: src/refer.ts [hunk 7]
```diff
@@ -35,39 +33,61 @@ function refer(this: any, options: any) {
...
 
+  async function actAcceptEntry(this: any, msg: any) {
+    const seneca = this
+
+    const entry = await seneca.entity('refer/entry').load$({ key: msg.key })
+
+    if (!entry) {
+      return {
+        ok: false,
+        error: 'No entry found with this key',
```
Comment (rjrodger) [inline src/refer.ts:48]: use the convention:
```
{ ok:false, why:'entry-unknown', details: { ... } }
```
* why should a "code", not human text
* details are optional
*

File: src/refer.ts [hunk 8]
```diff
@@ -81,32 +101,26 @@ function refer(this: any, options: any) {
     }
   }
 
-
   async function prepare(this: any) {
-    let seneca = this
+    const seneca = this
     await seneca.post('biz:refer,load:rules')
```
Comment (rjrodger) [inline src/refer.ts:106]: there is a bug I just found in seneca-msg-test that will prevent this from working - make sure to use the latest version

**Review by rjrodger — CHANGES_REQUESTED** [review]

File: src/refer.ts (hunk 7 above)
Comment (LeonNimoy) [reply src/refer.ts]: Got it! Return updated.

File: src/refer.ts (hunk 8 above)
Comment (LeonNimoy) [reply src/refer.ts:106]: Understood, lib version updated.
